smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaLatestCommit.java

```diff
@@ -54,13 +55,11 @@ public KafkaLatestCommit(Vertx vertx, KafkaConnectorIncomingConfiguration config
             if (last == null || last < record.getOffset() + 1) {
                 offsets.put(key, record.getOffset() + 1);
                 map.put(key, new OffsetAndMetadata(record.getOffset() + 1, null));
-                consumer.commit(map)
-                        .subscribe().with(x -> future.complete(null), future::completeExceptionally);
-            } else {
-                future.complete(null);
+                consumer.commitAsync(map)
```

Comment:
Perfect! I was previously curious why a reactive communication layer wasn't making use of an async API call.